src/kernel/code_page.zig

```diff
@@ -0,0 +1,88 @@
+const std = @import("std");
+const cp437 = @import("code_page/cp437.zig");
+
+/// The code page namespace
+pub const CodePage = struct {
+    /// The different code pages
+    pub const CodePages = enum {
+        /// Code page 437, the original IBM PC character set. Also known as OEM-US.
+        CP437,
+    };
+
+    /// The Error set for converting characters.
+    pub const Error = error{
+        /// The character to be converted is not part of the code page table.
+        InvalidChar,
+    };
+
+    ///
+    /// Convert a wide character (16-bits) to a code page.
+    ///
+    /// Arguments:
+    ///     IN code_page: CodePages - The code page to convert to.
+    ///     IN char: u16            - The character to convert.
+    ///
+    /// Return: u8
+    ///     The converted character.
+    ///
+    /// Error: Error
+    ///     error.InvalidChar - The character to be converted is not in the code page table.
+    ///
+    pub fn toCodePage(code_page: CodePages, char: u16) Error!u8 {
+        const table = switch (code_page) {
+            .CP437 => cp437.table,
+        };
```

Comment:
This could be made nice and abstract by adding a function to `CodePages` that returns the table. This could then be used in `toWideChar` as well.

Reply:
Not really sure what you mean. is it like:
```Zig
pub fn getTable(code_page: CodePage) []u8 {
    return switch (code_page) {
            .CP437 => cp437.table,
        };
}
```

Reply:
Exactly. That could go in the struct.

Reply:
what would the use be of it though? but can add it

Reply:
Well you have this logic in a couple of places so it would be good to extract it into the enum itself. Plus, encapsulation is nice :)

Reply:
I see now